src/pages/execution/workflowTabs/gantt-chart/ConductorTimeline.tsx

```diff
@@ -0,0 +1,274 @@
+    let subTaskMap = new Map<string, Series[]>();
+    parentTaskIds.forEach(taskId => {
+      let taskIdx = idToIndexMap.get(taskId)
+      let task = initialData[taskIdx];
+      subTaskMap.set(taskId, [task]);
+      let i = taskIdx+1;
+      let subTaskArr = subTaskMap.get(taskId);
+      while (i<initialData.length && initialData[i].parentTaskReferenceName === task.referenceTaskName){        
+        if (task.taskType === DO_WHILE){
+          let idx = subTaskArr.findIndex((subTask)=>subTask.referenceTaskName === initialData[i].referenceTaskName);
+          if (idx === -1){
+            subTaskArr.push(initialData[i]);
+          }else{
+            subTaskArr[idx] = {...subTaskArr[idx], data: [...subTaskArr[idx].data, initialData[i].data[0]]};
+          }
+        }else if (task.taskType === FORK_JOIN_DYNAMIC || task.taskType === FORK){
+          let newTime:Date = task.data[0].t2;
+          let oldTime:Date = subTaskArr[0].data[0].t2;
+          if (oldTime < newTime){
+            subTaskArr[0] = {...subTaskArr[0], data: [{...subTaskArr[0].data[0], t2: newTime}]}
+          }
+        }
+        i++;
+      } 
+      subTaskMap.set(taskId, subTaskArr);
+    })
+    return subTaskMap;
+  }, [collapsedData])
+  /** Map of Task IDs to their content when expanded */
+  const expandedTaskMap = useMemo<Map<string, Series[]>>(()=> {
+    let subTaskMap = new Map<string, Series[]>();
+    parentTaskIds.forEach(taskId => {
+      let taskIdx = idToIndexMap.get(taskId)
+      let task = initialData[taskIdx];
+      subTaskMap.set(taskId, [task]);
+      let i = taskIdx+1;
+      let subTaskArr = subTaskMap.get(taskId);
+      while (i<initialData.length && initialData[i].parentTaskReferenceName === task.referenceTaskName){
+        subTaskArr.push(initialData[i]); 
+        i++;
+      }    
+      subTaskMap.set(taskId, subTaskArr);
+    })
+    return subTaskMap;
+  }, [data])
+
+  const seriesMax:() => Date = ()=>{
+    let task:Series = series[series.length-1];
+    let idx:number = task.data.length-1;
+    return task.data[idx].t2;
+  }
+  const seriesMin:() => Date = ()=>{
+    let task:Series = series[0];
+    return task.data[0].w1 || task.data[0].t1;
+  }
+
+  
+
+  const [series, setSeries] = useState<Series[]>(collapsedData);  
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const [max, setMax] = useState(series && series.length? seriesMax():null);
+  const [min, setMin] = useState(series && series.length? seriesMin():null);
+  
+  useEffect(()=>{
+    setSeries(series.map(task =>{
+      task.data.forEach(span => {
+        span.styles = {
+          span: spanStyle(span.id, span.status),
+          waitSpan: spanStyle(span.id, span.status)
+        }
+      })
+      return task
+    })) 
+  }, [selectedTaskId])
+
+
+  function toggleAll(){
+    if (expanded){
+      let newData = [];
+      taskExpanded.forEach((value,key)=>value && taskExpanded.set(key,false));
```

Comment:
This might be problematic. `taskExpanded` is defined via a useMemo and should be immutable.